upcase_booklist_app/templates/books.html

```diff
@@ -0,0 +1,36 @@
+<!doctype html>
+<html>
+    <head>
+        <title>Curious Programmer</title>
+        <link rel="stylesheet" href="{{ url_for('static', filename='main.css') }}">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    </head>
+
+    <body> 
+        <div class="page-header">
+            <p class="nav-item">
+                <a href="{{ url_for('book_list') }}">Books</a>
```

Comment:
I'd recommend renaming the `book_list` template and view function to `books`, so it's similar to `authors`, and this is actually the general pattern you'll see in the real world (REST). See this reference:

https://guides.rubyonrails.org/routing.html#crud-verbs-and-actions

Note you're using the first and fourth type of routes in your application at this time. Also note in the fourth row that the url is generally `/books/:id`

Reply:
@paulghaddad The problem is that when I change the names I get errors bc Jinja confuses the view function name ("books") with the "books" dictionary that I pass in and later extract from.

In app:
```
@app.route("/books")
def book_list():
    """Show the list of books"""
    return render_template("books.html", books=books)

```
In books.html:

<div>
            {% for book_id, details in books.items() %} 
....